pkg/controller/ipam/validate_test.go

```diff
@@ -92,6 +92,102 @@ func TestEgressControllerValidateExternalIPPool(t *testing.T) {
 			},
 			expectedResponse: &admv1.AdmissionResponse{Allowed: true},
 		},
+		{
+			name: "CREATE operation with IP overlap should not be allowed",
+			request: &admv1.AdmissionRequest{
+				Name:      "foo",
+				Operation: "CREATE",
+				Object: runtime.RawExtension{Raw: marshal(copyAndMutateIPPool(testIPPool, func(pool *crdv1alpha2.IPPool) {
+					pool.Spec.IPRanges = append(pool.Spec.IPRanges, crdv1alpha2.SubnetIPRange{
+						IPRange: crdv1alpha2.IPRange{
+							CIDR: "192.168.3.0/26",
+						},
+						SubnetInfo: crdv1alpha2.SubnetInfo{
+							Gateway:      "192.168.3.1",
+							PrefixLength: 24,
+						},
+					})
+				}))},
+			},
+			expectedResponse: &admv1.AdmissionResponse{
+				Allowed: false,
+				Result: &metav1.Status{
+					Message: "IPRanges [192.168.3.10-192.168.3.20,192.168.3.0/26] overlap",
+				},
+			},
+		},
+		{
+			name: "CREATE operation with CIDR overlap with IP range should not be allowed",
```

Comment:
what's the difference between the above one and this one?

Reply:
1st one intersect while the 2nd one is contained. I added the latter as your review comment above was that the latter case is not covered by the validation. If you think that one of these is redundant, I can remove it.

Reply:
The two cases look good to me. I just didn't tell the difference from the name of the cases. Could we make the name a bit more explict to explain the difference?